### DISCUSSION TITLE
Issue with Custom Connector OAuth in Power BI Desktop

### DISCUSSION THREAD

**Discussion by hernerj** [microsoft/vscode-powerquery-sdk#221] [General]
Hello, 
I have been working on creating a custom connector to go against a basic API the uses OAuth flow. Everything seems to work ok except that I can't seem to make the pop-up sign in page actually prompt the user for any input. Every time the sign in button is clicked, it brings up a little window which automatically goes through the login process and pulls my credentials from "somewhere" before closing the window. The same thing also happens when you click the "Sign in as a different user" button. This means that I cannot use multiple accounts in my connector because it always automatically signs me in to the same account without allowing me to choose a different one. I have tried logging out of accounts in my browsers and power bi as well as clearing my browser and power bi caches to no avail. The same behavior seems to happen when I checked the "Use default browser" setting in the browser authentication section in power bi. Has anyone else experienced or solved similar issues?
Thanks!

**Comment by ptyx507x** [comment]
Hey,

This is more than likely caused by something happening with your StartLogin and how it interacts with the browser.

Have you tried using something like Fiddler to capture the traces of the calls made by your computer against your REST API to make sure that the correct call is being made? if the correct call is being made, is the correct response being received? Is there perhaps some interaction happening at the browser level where it automatically signs you in? Could it be that there's a missing query parameter for the StartLogin Url that forces the "Sign in" to occur and never go through any sort of auto-login?

One workaround in Power BI Desktop today is to simply "Clear credentials". That effectively removes the credentials that you have in your Power BI Desktop and you can sign in with a new account.

**Reply by hernerj** [reply]
When you say "Clear Credentials" are you talking about the data source settings section? I've tried deleting everything in there but it still automatically signs me back in to the same account when I connect to the data source again.

As far as capturing the login process with fiddler, we have done that. Everything seems to look ok. I've also tried adding a parameter "prompt = login" to the request made in the start login function and that seemed to have no effect. On the Azure B2C side we do not believe we have any settings that would cause automatic login. 

Below is a basic rundown of the requests I see in fiddler when I hit login in power bi desktop:

GET  (200) /<b2c_tenant>.onmicrosoft.com/<sign_in_policy>/oauth2/logout
GET  (302) /<b2c_tenant>.onmicrosoft.com/<sign_in_policy>/oauth2/v2.0/authorize?client_id=<cliend_id>&response_type=code&response_mode=query&code_challenge_method=S256&code_challenge=<challenge>&state=<state>&prompt=login&redirect_uri=https%3A%2F%2Foauth.powerbi.com%2Fviews%2Foauthredirect.html&scope=openid%20offline_access
GET  (200) /common/oauth2/authorize?client_id=<b2c_client_id>&redirect_uri=<b2c_redirect_uri>&response_type=code&scope=openid&response_mode=form_post&nonce=<nonce>&state=<state_props>
POST (302) /<b2c_tenant>.onmicrosoft.com/oauth2/authresp
POST (200) /<b2c_tenant>.onmicrosoft.com/<sign_in_policy>/oauth2/v2.0/token

**Comment by sea1jxr** [comment]
AAD is trying to help you because the user is already signed in.  Try using prompt=select_account instead to force a login prompt to show up even for authenticated sessions.

**Comment by hernerj** [comment]
We solved this issue by changing the logout url. The url we use in our web app was apparently not actually working for power bi so we were never actually logging out even though it appeared like we were. Once we found the correct url to use for logout, we were prompted for login at last. Thanks for the responses here.

**Reply by ptyx507x** [reply]
Glad to hear this!
